src/core/tokenize/match.js

```diff
@@ -21,7 +21,12 @@ export function _matchGrammar (text, tokenList, grammar, startNode, startPos, re
 
 	for (const token in grammar) {
 		const tokenValue = grammar[token];
-		if (!grammar.hasOwnProperty(token) || token.startsWith('$') || !tokenValue) {
+		if (
+			!grammar.hasOwnProperty(token) ||
+			token.startsWith('$') ||
+			!tokenValue ||
+			typeof tokenValue === 'function'
```

Comment:
Functional tokens (`$inside` for now) are handled on L170, and we should ignore them in all other cases.